integrationtests/src/test/java/org/axonframework/integrationtests/commandhandling/AbstractAggregateMemberTest.java

```diff
@@ -0,0 +1,233 @@
+package org.axonframework.integrationtests.commandhandling;
+
+
+import org.axonframework.commandhandling.CommandHandler;
+import org.axonframework.eventsourcing.EventSourcingHandler;
+import org.axonframework.modelling.command.*;
+import org.axonframework.spring.stereotype.Aggregate;
+import org.axonframework.test.aggregate.AggregateTestFixture;
+import org.axonframework.test.aggregate.FixtureConfiguration;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class AbstractAggregateMemberTest {
+    private FixtureConfiguration<FactoryAggregate> fixture;
+    private String factoryId = "factoryId";
+
+    @BeforeEach
+    public void setUp(){
+        fixture = new AggregateTestFixture(FactoryAggregate.class);
+    }
+
+    @Test
+    public void testInitFactoryAggregate_ShouldHandleCommandNormally(){
+        fixture.givenNoPriorActivity()
+                .when(new CreateFactoryCommand(factoryId))
+                .expectEvents(new FactoryCreatedEvent(factoryId));
+    }
+
+    @Test
+    public void testEmployees_ShouldBeAbleToHandleCommand(){
+        fixture.givenCommands(new CreateFactoryCommand(factoryId))
+                .when(new SomethingCommand(factoryId, "employeeId"))
+                .expectEvents(new SomethingEvent(factoryId, "employeeId"));
+    }
+
+    @Test
+    public void testManagers_ShouldBeAbleToHandleCommand(){
+        fixture.givenCommands(new CreateFactoryCommand(factoryId))
+                .when(new SomethingCommand(factoryId, "managerId"))
+                .expectEvents(new SomethingEvent(factoryId, "managerId"));
+    }
+
+
+    private static abstract class Person {
+        @EntityId
+        public String personId;
+    }
+
+    private static class Employee extends Person {
+
+        public Employee(){
+        }
+
+        public Employee(String personId){
+            this.personId = personId;
+        }
+
+        @CommandHandler
+        public void handle(SomethingCommand cmd){
+            AggregateLifecycle.apply(new SomethingEvent(cmd.factoryId, cmd.personId));
```

Comment:
Two small and nit-picky suggestions. 

As we throughout use getters in virtually all scenarios, even for inner classes, I'd like to see the usage of `cmd.getFactoryId()` instead of the direct field reference.

Secondly, the event publication is entirely identical between the `Employee` and `Manager` entity implementations.
To make it just that bit clearer, I think it would be got to either (1) have separate employee and manager events or (2) add some Employee/Manager specific static value to the published `SomethingEvent`.